### PR TITLE
Add configurable parameters

### PR DESCRIPTION
**Pull request by matthewbartos**
Solves: #9 

## Changes:
- [x] Introduced `intersectionType`, `acceleration`, `max speed`, `spawn rate` as configurable parameters
- [x] Added wireframes for intersection types

### CODE REVIEW

File: sz/Traffic/src-gen/traffic/ReLogoLink.java [hunk 1]
```diff
@@ -296,5 +296,68 @@ public boolean isUserLinkQ(Object o){
 		return userLink(oneEnd.getWho(), otherEnd.getWho());
 	}
 
+	/**
+	 * Returns the value of the global variable intersectionType.
+	 *
+	 * @return the value of the global variable intersectionType
+	 */
+	@ReLogoBuilderGeneratedFor("global: intersectionType")
```
Comment (matthewbartos) [inline sz/Traffic/src-gen/traffic/ReLogoLink.java:304]: Do you think that we need to include these files from `src-gen` in our commits?

File: sz/Traffic/src/traffic/relogo/UserObserver.groovy [hunk 2]
```diff
@@ -16,11 +16,11 @@ class UserObserver extends ReLogoObserver{
 		clearAll()
 		setDefaultShape(UserTurtle, "car")
 		
-		def w = (UserPatch.laneWidth + 1) / 2
+		def width = (UserPatch.laneWidth + 1) / 2
 		def d = 15
```
Comment (matthewbartos) [inline sz/Traffic/src/traffic/relogo/UserObserver.groovy:20]: What does the `d` do here? 🙂

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy [hunk 3]
```diff
@@ -16,7 +16,21 @@ class UserPatch extends ReLogoPatch{
 	def setColor() {
 		def x = getPxcor()
 		def y = getPycor()
-		
+
+		if (intersectionType == "p2pIntersection") {
```
Comment (matthewbartos) [inline sz/Traffic/src/traffic/relogo/UserPatch.groovy:20]: `switch` is not working 🙁 
```
switch(intersectionType) {
	case "p2pIntersection":
		colorP2pIntersection(x, y)
		break

	case "priority":
		colorPriority(x, y)
		break

	case "trafficLights":
		colorTrafficLights(x, y)
		break

	case "roundabout":
		colorRoundabout(x, y)
		break

	default:
		colorP2pIntersection(x, y)
		break
	}
```

File: sz/Traffic/src-gen/traffic/ReLogoLink.java (hunk 1 above)
Comment (hejmsdz) [reply sz/Traffic/src-gen/traffic/ReLogoLink.java:304]: If Repast is able to regenerate these files after pulling, then we can skip them, but I'm not sure if that's the case.

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy (hunk 3 above)
Comment (hejmsdz) [reply sz/Traffic/src/traffic/relogo/UserPatch.groovy:20]: 🤷‍♂

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy [hunk 4]
```diff
@@ -16,7 +16,21 @@ class UserPatch extends ReLogoPatch{
...
+			colorP2pIntersection(x, y)
+		} else if (intersectionType == "priority") {
+			colorPriority(x, y)
+		} else if (intersectionType == "trafficLights") {
+			colorTrafficLights(x, y)
+		}else if (intersectionType == "roundabout") {
```
Comment (hejmsdz) [inline sz/Traffic/src/traffic/relogo/UserPatch.groovy:26]:
```suggestion
		} else if (intersectionType == "roundabout") {
```

File: sz/Traffic/src/traffic/relogo/UserObserver.groovy (hunk 2 above)
Comment (hejmsdz) [reply sz/Traffic/src/traffic/relogo/UserObserver.groovy:20]: This is the distance from origin to the location of each spawn point.

File: sz/Traffic/src/traffic/relogo/PoissonStream.groovy [hunk 5]
```diff
@@ -12,7 +12,7 @@ import repast.simphony.relogo.schedule.Setup
 import traffic.ReLogoTurtle
 
 class PoissonStream extends ReLogoTurtle {
-	def rate = 0.2
+	def poissonStreamRate = 0.2
```
Comment (hejmsdz) [inline sz/Traffic/src/traffic/relogo/PoissonStream.groovy]: In order to use global variable from the slider, shouldn't you get rid of this property?

File: sz/Traffic/src-gen/traffic/ReLogoLink.java (hunk 1 above)
Comment (matthewbartos) [reply sz/Traffic/src-gen/traffic/ReLogoLink.java:304]: Nope, we need to keep them 🙁

File: sz/Traffic/src/traffic/relogo/UserObserver.groovy (hunk 2 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/UserObserver.groovy:20]: 👍

File: sz/Traffic/src/traffic/relogo/UserPatch.groovy (hunk 4 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/UserPatch.groovy:26]: Yeah, Eclipse doesn't have autoformat 😕

File: sz/Traffic/src/traffic/relogo/PoissonStream.groovy (hunk 5 above)
Comment (matthewbartos) [reply sz/Traffic/src/traffic/relogo/PoissonStream.groovy]: True, fixing

**Pull request merged**